src/types/wrappers/MaxMinDrivingForceModel.jl

```diff
@@ -0,0 +1,307 @@
+        dg0s
+    ]
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Get the equality constraint lhs of `model`.
+"""
+function Accessors.stoichiometry(model::MaxMinDrivingForceModel)
+    var_ids = get_unmangled_variables(model)
+
+    # set proton and water equality constraints
+    num_proton_water = length(model.proton_ids) + length(model.water_ids)
+    proton_water_mat = spzeros(num_proton_water, n_variables(model))
+    idxs = indexin([model.proton_ids; model.water_ids], var_ids)
+    for (i, j) in enumerate(idxs)
+        isnothing(j) && throw(error("Water or proton ID not found in model."))
+        proton_water_mat[i, j] = 1.0
+    end
+
+    # constant concentration constraints
+    const_conc_mat = spzeros(length(model.constant_concentrations), n_variables(model))
+    ids = collect(keys(model.constant_concentrations))
+    idxs = indexin(ids, var_ids)
+    for (i, j) in enumerate(idxs)
+        isnothing(j) &&
+            throw(DomainError(ids[j], "Constant metabolite ID not found in model."))
+        const_conc_mat[i, j] = 1.0
+    end
+
+    # add the relative bounds
+    const_ratio_mat = spzeros(length(model.concentration_ratios), n_variables(model))
+    for (i, (mid1, mid2)) in enumerate(keys(model.concentration_ratios))
+        idxs = indexin([mid1, mid2], var_ids)
+        any(isnothing.(idxs)) &&
+            throw(DomainError((mid1, mid2), "Metabolite ratio pair not found in model."))
+        const_ratio_mat[i, first(idxs)] = 1.0
+        const_ratio_mat[i, last(idxs)] = -1.0
+    end
+
+    # add ΔG relationships
+    dgrs = spdiagm(ones(length(reactions(model))))
+    S = stoichiometry(model.inner)
+    stoich_mat = -(model.R * model.T) * S'
+    dg_mat = [spzeros(n_reactions(model)) stoich_mat dgrs]
+
+    return [
+        proton_water_mat
+        const_conc_mat
+        const_ratio_mat
+        dg_mat
+    ]
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Get the simple variable bounds of `model`.
+"""
+function Accessors.bounds(model::MaxMinDrivingForceModel)
+    var_ids = get_unmangled_variables(model)
+
+    lbs = fill(-model.max_dg_bound, n_variables(model))
+    ubs = fill(model.max_dg_bound, n_variables(model))
+
+    # mmdf must be positive for problem to be feasible (it is defined as -ΔG)
+    lbs[1] = 0.0
+    ubs[1] = 1000.0
+
+    # log concentrations
+    lbs[2:(1+n_metabolites(model))] .= log(model.concentration_lb)
+    ubs[2:(1+n_metabolites(model))] .= log(model.concentration_ub)
+
+    # need to make special adjustments for the constants
+    idxs = indexin([model.proton_ids; model.water_ids], var_ids)
+    lbs[idxs] .= -1.0
+    ubs[idxs] .= 1.0
+
+    # ΔG for each reaction can be any sign, but that is filled before default
+
+    return (lbs, ubs)
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+A helper function that returns the reaction ids that are active. Active reaction
+have thermodynamic data AND a flux bigger than `small_flux_tol` AND are not
+ignored.
+"""
+_get_active_reaction_ids(model::MaxMinDrivingForceModel) = filter(
```

Comment:
underscores are obsolete, `get_` prefix is imperative...

...move this to Internal, like `mmdf_active_reaction_ids` or so